presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -90,54 +91,65 @@ public PlanNode optimize(PlanNode plan, Session session, Map<Symbol, Type> types
         requireNonNull(symbolAllocator, "symbolAllocator is null");
         requireNonNull(idAllocator, "idAllocator is null");
         if (SystemSessionProperties.isOptimizeHashGenerationEnabled(session)) {
-            PlanWithProperties result = plan.accept(new Rewriter(idAllocator, symbolAllocator, types), new HashSymbolPreference());
+            PlanWithProperties result = plan.accept(new Rewriter(idAllocator, symbolAllocator, types), new HashSymbols());
             return result.getNode();
         }
         return plan;
     }
 
-    private static class HashSymbolPreference
+    private static class HashSymbols
     {
-        private final Set<List<Symbol>> preferredHashSymbols;
+        private final Set<List<Symbol>> hashSymbols;
```

Comment:
Got it. The name of the variable was throwing me off. I thought it was meant to contain the hash symbols (i.e., the output of the hash function), not the inputs. We should rename it for clarity. Maybe something like `fieldsToHash`, `hashInputs` or something along those lines.